feat: contribution guide (#7)
Moong-glE/my-community-backend: CONTRIBUTING.md

# Team Moon-glE Contributing Guidelines

Please follow these guidelines when you are trying to contribute to MCIP software.

## Language

> We **_prefer English over Korean_** due to its globally high usage and impact. So it is strongly recommended to use English throughout your development routine. Don't worry about your bad English. We don't judge any kind of your grammar or language structure including ours. However, it is important to make sure that each sentence you wrote should be easily understandable and has an exact atomic meaning without ambiguity.

Reviewer: I prefer Korean over English

Author: Modified~!

Generally there are two acceptable styles of commit message.

### commit type: what you did in present-tense, imperative-style

Refer to [here](http://karma-runner.github.io/0.10/dev/git-commit-msg.html) and [here](https://seesparkbox.com/foundry/semantic_commit_messages) for more information.

**Available types**

- chore (updating grunt tasks etc; no production code change)
- docs (changes to documentation)
- feat (new feature)
- fix (bug fix)
- refactor (refactoring production code)
- style (formatting, missing semi colons, etc; no code change)
- test (adding missing tests, refactoring tests; no production code change)

**Examples**

- `chore: add Oyster build script`
- `docs: explain hat wobble`
- `feat: add beta sequence`
- `fix: remove broken confirmation message`
- `refactor: share logic between 4d3d3d3 and flarhgunnstow`
- `style: convert tabs to spaces`
- `test: ensure Tayne retains clothing`

> You can also use other types like `misc:`, `design:`, `types:`, `deps:` or whatever that fit well with your works.

### Start with a present-tense verb in imperative-style with a first character in uppercase

**Examples**

- `Update README.md`
- `Add assets`

## Pull Requests

> 

Reviewer: 풀리퀘 가이드도 추가해주세용

Author: 추가해써용~

# Team Moon-glE Contributing Guidelines

Please follow these guidelines when you are trying to contribute to MCIP software.

## Language

We **_prefer English and Korean_** due to all of our team member is Korean, we prefer Korean. So it is strongly recommended to use Korean. But it needs to conversate in gobally, we also prefer the English.

## Code Formatting

> Keep your codes clean and consistent using code formatters and linters. Most of the time formatting and linting would already have been set by default in each repository. And sometimes you should use the specific code editor to match the configuration.

Reviewer 2: `Keep your codes clean and consistent using code formatters and linters. Most of the time formatting and linting would already have been set by default in each repository. And sometimes you should use the specific code editor to match the configuration.` 
Is too generalized.

`Keep your codes clean and consistent using code formatters and linters. we prefer to use basic PEP8 format.` seems to be appropriate for our project

# Team Moon-glE Contributing Guidelines

> Please follow these guidelines when you are trying to contribute to MCIP software.

Reviewer 3: ```suggestion
Please follow these guidelines when you are trying to contribute to **My Community** software.
```

## Language

> We **_prefer English and Korean_** due to all of our team member is Korean, we prefer Korean. So it is strongly recommended to use Korean. But it needs to conversate in gobally, we also prefer the English.

Reviewer 3: ```suggestion
We **_prefer English and Korean_** due to all of our team member is Korean, we prefer Korean. So it is strongly recommended to use Korean. But it needs to conversate in globally, we also prefer the English.
```

## Code Formatting

Keep your codes clean and consistent using code formatters and linters. Most of the time formatting and linting would already have been set by default in each repository. And sometimes you should use the specific code editor to match the configuration.

## Git commit messages

Generally there are two acceptable styles of commit message.

### commit type: what you did in present-tense, imperative-style

> Refer to [here](http://karma-runner.github.io/0.10/dev/git-commit-msg.html) and [here](https://seesparkbox.com/foundry/semantic_commit_messages) for more information.

Reviewer 2: 그래서 이거는 그냥 contribution guide에대한 설명이라 
`Generally there are two acceptable styles of commit message.`
이거랑
`Refer to [here](http://karma-runner.github.io/0.10/dev/git-commit-msg.html) and [here](https://seesparkbox.com/foundry/semantic_commit_messages) for more information.`
이거는 지워줘도될듯

**Available types**

- chore (updating grunt tasks etc; no production code change)
- docs (changes to documentation)
- feat (new feature)
- fix (bug fix)
- refactor (refactoring production code)
- style (formatting, missing semi colons, etc; no code change)
- test (adding missing tests, refactoring tests; no production code change)

> **Examples**

Reviewer 3: 여기 다 대문자로 고쳐주세요!

예) `chore: Add Oyster build script`

- `chore: add Oyster build script`
- `docs: explain hat wobble`
- `feat: add beta sequence`
- `fix: remove broken confirmation message`
- `refactor: share logic between 4d3d3d3 and flarhgunnstow`
- `style: convert tabs to spaces`
- `test: ensure Tayne retains clothing`

> You can also use other types like `misc:`, `design:`, `types:`, `deps:` or whatever that fit well with your works.

### Start with a present-tense verb in imperative-style with a first character in uppercase

> **Examples**

Reviewer 2: 두가지 커밋타입이 혼재되어있음!

둘중하나 지우고 하나로 택해야하는데 나는 전자스타일
- `chore: add Oyster build script`
- `docs: explain hat wobble`

선호함
@RyuJaesang @GunHee5719 @jhcheon1000

Reviewer: 나도 전자 선호 깰꿈쓰

Reviewer 3: 나도 전자

Reviewer 3: 그리고 나는 개인적으로 커밋타입 너무 많은거 비추함
너무 많으면 어떤거 써야할지도 잘 모르고 중구난방되는 느낌이라??
`style` 요런 타입도 `chore` 같은 타입으로 대체할 수 있다고 생각함

`chore`, `feat`, `fix`, `refactor`, `test` 정도로 나는 추천!

Author: chore, feat, fix, refactor, test, docs 총 6개로 땅땅

> - `Update README.md`

Reviewer 3: 여기 앞에 타입 붙여주시죠!

예) `chore: Update README.md`

- `Add assets`

## Pull Requests
- There are 2 types of _PULL_REQUEST_TEMPLATE_, **feature-template** and **bug-template**
- Follow our _Pull request template_ **STRICTLY** when you hang out the pull request
- **feature-template**

> - `Function description

Reviewer 3: ```suggestion
- `Function description`
```
